src/firefly/js/visualize/ui/ExpandedOptionsPopup.jsx

```diff
@@ -3,273 +3,137 @@
-import {useStoreConnector} from '../../ui/SimpleComponent';
-import CompleteButton from '../../ui/CompleteButton.jsx';
+import {isEqual} from 'lodash';
+import React, {useEffect, useState} from 'react';
+import {dispatchShowDialog} from '../../core/ComponentCntlr.js';
+import {TABLE_FILTER, TABLE_FILTER_SELROW, TABLE_LOADED, TABLE_SORT} from '../../tables/TablesCntlr.js';
+import {getTblById, watchTableChanges} from '../../tables/TableUtil.js';
+import {TablePanel} from '../../tables/ui/TablePanel.jsx';
+import {CompleteButton} from '../../ui/CompleteButton.jsx';
 import DialogRootContainer from '../../ui/DialogRootContainer.jsx';
+import {HelpIcon} from '../../ui/HelpIcon.jsx';
 import {PopupPanel} from '../../ui/PopupPanel.jsx';
-import {visRoot, dispatchChangeActivePlotView, dispatchDeletePlotView} from '../ImagePlotCntlr.js';
-import {primePlot} from '../PlotViewUtil.js';
+import {useStoreConnector} from '../../ui/SimpleComponent.jsx';
 import {
-    getMultiViewRoot, dispatchReplaceViewerItems,
-    EXPANDED_MODE_RESERVED, IMAGE, dispatchAddViewer, dispatchAddViewerItems
-} from '../MultiViewCntlr.js';
-import {dispatchShowDialog} from '../../core/ComponentCntlr.js';
-import {TablePanel} from '../../tables/ui/TablePanel';
-import {dispatchTableAddLocal, TABLE_SORT} from '../../tables/TablesCntlr';
-import {getTblById, getTblInfoById, processRequest, watchTableChanges} from '../../tables/TableUtil';
-import {getFormattedWaveLengthUnits, getPlotViewAry, getPlotViewById, isPlotViewArysEqual} from '../PlotViewUtil';
-import {PlotAttribute} from '../PlotAttribute';
-import {TABLE_LOADED, TABLE_FILTER, TABLE_FILTER_SELROW} from '../../tables/TablesCntlr';
-import {getViewerItemIds} from '../MultiViewCntlr';
-import {HelpIcon} from '../../ui/HelpIcon';
-import {SelectInfo} from '../../tables/SelectInfo';
-
+    EXPANDED_OPTIONS_DIALOG_ID, HIDDEN, IMAGE_VIEW_TABLE_ID,
+    getCombinedItemIds, getPvAryInViewer, isOnlyPlotSortOutOfSync,
+    isPlotFilterOrSortOutOfSync, makeImViewDisplayModel, removeImageViewDisplaySelected, syncSort,
+    updateImageViewDisplay, addExpandFilterSyncWatcher
+} from '../ImViewFilterDisplay.js';
+import {dispatchAddViewer, dispatchAddViewerItems, EXPANDED_MODE_RESERVED, IMAGE} from '../MultiViewCntlr.js';
+import {deleteAllFailedPlots, isPlotViewArysEqual} from '../PlotViewUtil.js';
+import {ListViewButton} from './Buttons.jsx';
+
+
+export function ViewOptionsButton({ title, viewerId=EXPANDED_MODE_RESERVED }) {
+    const outOfSyncTip= 'Click to re-add filters after image change';
+    const inSyncTip= 'Choose which images to show';
+    const outOfSync= isPlotFilterOrSortOutOfSync(viewerId);
+    const onlySortOutOfSync= isOnlyPlotSortOutOfSync(viewerId);
 
-const TABLE_ID = 'active-image-view-list-table';
+    useEffect(() => {
+        addExpandFilterSyncWatcher();
+    }, []);
 
-const HIDDEN='_HIDDEN';
+    useEffect(() => {
+        onlySortOutOfSync && syncSort(viewerId);
+    },[onlySortOutOfSync]);
```

Comment:
I don't think this component re-render.  This can be apply in the previous useEffect.